### PR TITLE
platformio 4.0.0

### PR DESCRIPTION
**Pull request by ivankravets**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

[Migration Guide from 3.0 to 4.0](http://docs.platformio.org/page/migration.html).

* [PlatformIO Plus Goes Open Source](https://community.platformio.org/t/platformio-plus-goes-open-source-improving-embedded-development-community-worldwide/8240/4)

  - Built-in [PIO Unified Debugger](http://docs.platformio.org/page/plus/debugging.html)
  - Built-in [PIO Unit Testing](http://docs.platformio.org/page/plus/unit-testing.html)

* **Project Configuration**

  - New project configuration parser with a strict options typing ([API](https://github.com/platformio/platformio-core/blob/develop/platformio/project/options.py))
  - Unified workspace storage ([workspace_dir](http://docs.platformio.org/page/projectconf/section_platformio.html#workspace-dir) -> ``.pio``) for PlatformIO Build System, Library Manager, and other internal services ([issue #1778](https://github.com/platformio/platformio-core/issues/1778))
  - Share common (global) options between project environments using [[env]](http://docs.platformio.org/page/projectconf/section_env.html#global-scope-env) section ([issue #1643](https://github.com/platformio/platformio-core/issues/1643))
  - Include external configuration files with [extra_configs](http://docs.platformio.org/page/projectconf/section_platformio.html#extra-configs) option ([issue #1590](https://github.com/platformio/platformio-core/issues/1590))
  - Custom project ``***_dir`` options declared in [platformio](http://docs.platformio.org/page/projectconf/section_platformio.html) section have higher priority than [Environment variables](http://docs.platformio.org/page/envvars.html)
  - Added support for Unix shell-style wildcards for [monitor_port](http://docs.platformio.org/page/projectconf/section_env_monitor.html#monitor-port) option ([issue #2541](https://github.com/platformio/platformio-core/issues/2541))
  - Added new [monitor_flags](http://docs.platformio.org/page/projectconf/section_env_monitor.html#monitor-flags) option which allows passing extra flags and options to [platformio device monitor](http://docs.platformio.org/page/userguide/cmd_device.html#cmd-device-monitor) command ([issue #2165](https://github.com/platformio/platformio-core/issues/2165))
  - Added support for [PLATFORMIO_DEFAULT_ENVS](http://docs.platformio.org/page/envvars.html#envvar-PLATFORMIO_DEFAULT_ENVS) system environment variable ([issue #1967](https://github.com/platformio/platformio-core/issues/1967))
  - Added support for [shared_dir](http://docs.platformio.org/page/projectconf/section_platformio.html#shared-dir) where you can place an extra files (extra scripts, LD scripts, etc.) which should be transferred to a [PIO Remote](http://docs.platformio.org/page/plus/pio-remote.html) machine

* **Library Management**

  - Switched to workspace ``.pio/libdeps`` folder for project dependencies instead of ``.piolibdeps``
  - Save libraries passed to [platformio lib install](http://docs.platformio.org/page/userguide/lib/cmd_install.html) command into the project dependency list ([lib_deps](http://docs.platformio.org/page/projectconf/section_env_library.html#lib-deps)) with a new ``--save`` flag ([issue #1028](https://github.com/platformio/platformio-core/issues/1028))
  - Install all project dependencies declared via [lib_deps](http://docs.platformio.org/page/projectconf/section_env_library.html#lib-deps) option using a simple [platformio lib install](http://docs.platformio.org/page/userguide/lib/cmd_install.html) command ([issue #2147](https://github.com/platformio/platformio-core/issues/2147))
  - Use isolated library dependency storage per project build environment ([issue #1696](https://github.com/platformio/platformio-core/issues/1696))
  - Look firstly in built-in library storages for a missing dependency instead of PlatformIO Registry ([issue #1654](https://github.com/platformio/platformio-core/issues/1654))
  - Override default source and include directories for a library via [library.json](http://docs.platformio.org/page/librarymanager/config.html) manifest using ``includeDir`` and ``srcDir`` fields
  - Fixed an issue when library keeps reinstalling for non-latin path ([issue #1252](https://github.com/platformio/platformio-core/issues/1252))
  - Fixed an issue when [lib_compat_mode = strict](http://docs.platformio.org/page/librarymanager/ldf.html#ldf-compat-mode) does not ignore libraries incompatible with a project framework

* **Build System**

  - Switched to workspace ``.pio/build`` folder for build artifacts instead of ``.pioenvs``
  - Switch between [Build Configurations](http://docs.platformio.org/page/projectconf/build_configurations.html) (``release`` and ``debug``) with a new project configuration option [build_type](http://docs.platformio.org/page/projectconf/section_env_build.html#build-type)
  - Custom [platform_packages](http://docs.platformio.org/page/projectconf/section_env_general.html#platform) per a build environment with an option to override default ([issue #1367](https://github.com/platformio/platformio-core/issues/1367))
  - Print platform package details, such as version, VSC source and commit ([issue #2155](https://github.com/platformio/platformio-core/issues/2155))
  - Control a number of parallel build jobs with a new [-j, --jobs](http://docs.platformio.org/page/userguide/cmd_run.html#cmdoption-platformio-run-j) option
  - Override default ["platformio.ini" (Project Configuration File)](https://docs.platformio.org/page/projectconf.html) with a custom using ``-c, --project-conf`` option for [platformio run](http://docs.platformio.org/page/userguide/cmd_run.html), [platformio debug](http://docs.platformio.org/page/userguide/cmd_debug.html), or [platformio test](http://docs.platformio.org/page/userguide/cmd_test.html) commands ([issue #1913](https://github.com/platformio/platformio-core/issues/1913))
  - Override default development platform upload command with a custom [upload_command](http://docs.platformio.org/page/projectconf/section_env_upload.html#upload-command) ([issue #2599](https://github.com/platformio/platformio-core/issues/2599))
  - Configure a shared folder for the derived files (objects, firmwares, ELFs) from a build system using [build_cache_dir](http://docs.platformio.org/page/projectconf/section_platformio.html#build-cache-dir) option ([issue #2674](https://github.com/platformio/platformio-core/issues/2674))
  - Fixed an issue when ``-U`` in ``build_flags`` does not remove macro previously defined via ``-D`` flag ([issue #2508](https://github.com/platformio/platformio-core/issues/2508))

* **Infrastructure**

  - Python 3 support ([issue #895](https://github.com/platformio/platformio-core/issues/895))
  - Significantly speedup back-end for PIO Home. It works super fast now!
  - Added support for the latest Python "Click" package (CLI) ([issue #349](https://github.com/platformio/platformio-core/issues/349))
  - Added options to override default locations used by PlatformIO Core ([core_dir](http://docs.platformio.org/page/projectconf/section_platformio.html#core-dir), [globallib_dir](http://docs.platformio.org/page/projectconf/section_platformio.html#globallib-dir), [platforms_dir](http://docs.platformio.org/page/projectconf/section_platformio.html#platforms-dir), [packages_dir](http://docs.platformio.org/page/projectconf/section_platformio.html#packages-dir), [cache_dir](http://docs.platformio.org/page/projectconf/section_platformio.html#cache-dir)) ([issue #1615](https://github.com/platformio/platformio-core/issues/1615))
  - Removed line-buffering from [platformio run](http://docs.platformio.org/page/userguide/cmd_run.html) command which was leading to omitting progress bar from upload tools ([issue #856](https://github.com/platformio/platformio-core/issues/856))
  - Fixed numerous issues related to "UnicodeDecodeError" and international locales, or when project path contains non-ASCII chars ([issue #143](https://github.com/platformio/platformio-core/issues/143), [issue #1342](https://github.com/platformio/platformio-core/issues/1342), [issue #1959](https://github.com/platformio/platformio-core/issues/1959), [issue #2100](https://github.com/platformio/platformio-core/issues/2100))

* **Integration**

  - Support custom CMake configuration for CLion IDE using ``CMakeListsUser.txt`` file
  - Fixed an issue with hardcoded C standard version when generating project for CLion IDE ([issue #2527](https://github.com/platformio/platformio-core/issues/2527))
  - Fixed an issue with Project Generator when an include path search order is inconsistent to what passed to the compiler ([issue #2509](https://github.com/platformio/platformio-core/issues/2509))
  - Fixed an issue when generating invalid "Eclipse CDT Cross GCC Built-in Compiler Settings" if a custom [PLATFORMIO_CORE_DIR](http://docs.platformio.org/page/envvars.html#envvar-PLATFORMIO_CORE_DIR) is used ([issue #806](https://github.com/platformio/platformio-core/issues/806))

* **Miscellaneous**

  - Deprecated ``--only-check`` PlatformIO Core CLI option for "update" sub-commands, please use ``--dry-run`` instead
  - Fixed "systemd-udevd" warnings in [99-platformio-udev.rules](http://docs.platformio.org/page/faq.html#platformio-udev-rules) ([issue #2442](https://github.com/platformio/platformio-core/issues/2442))
  - Fixed an issue when package cache (Library Manager) expires too fast ([issue #2559](https://github.com/platformio/platformio-core/issues/2559))

### CODE REVIEW

File: Formula/platformio.rb [hunk 1]
```diff
@@ -13,16 +13,16 @@ class Platformio < Formula
     sha256 "15d1add995b55416e0d42dad9e884e099416a14650e975e90ec91ff6a418a5bf" => :sierra
   end
 
-  depends_on "python@2" # does not support Python 3
+  depends_on "python" # Homebrew's Python 2.7.x (if installed) otherwise the macOS system Python
```
Comment (fxcoudert) [inline Formula/platformio.rb]: This comment is inaccurate. `python` formula is Homebrew's Python 3. Does this build with Python 3 or not?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/platformio.rb (hunk 1 above)
Comment (ivankravets) [reply Formula/platformio.rb]: What do you mean? I used your docs https://docs.brew.sh/Homebrew-and-Python

Yes, PlatformIO 4.0 is compatible with Python 2 and Python 3

Comment (ivankravets) [reply Formula/platformio.rb]: I removed that comment. Please update docs.

**Comment by fxcoudert** [comment]
Thanks @ivankravets

**Pull request closed without merging**